### ISSUE TITLE
[ON HOLD — #23546][$1000] Chat - Several images at once are skipped when switching attachments with arrows

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22318]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open staging.new.expensify.com with Safari
2. Login in any account
3. Go to any chat with multiple attachments
4. Select any attachment
5. Switch attachments with on-screen arrows

## Expected Result:
Only the next attachment is loaded by clicking on the on-screen arrow

## Actual Result:
Several images at once are skipped when switching attachments with on-screen arrows

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.37.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/788b44a2-3d0e-4ab4-acab-9894019f42d4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f9c0565b91f7e6d4</li>
        <li>Upwork Job ID: 1677033938685763584</li>
        <li>Last Price Increase: 2023-08-03</li>
    </ul>
</details>

**Comment by CortneyOfstad** [comment]
Was able to recreate, so going to get eyes on this 👍

**Comment by univalchemist** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Several images at once are skipped when switching attachments with arrows

### What is the root cause of that problem?
The prop `maxToRenderPerBatch` of FlastList cause this issue. As mentioned in official React Native documentation, this prop `maxToRenderPerBatch` is a `VicualizedList` prop that can be passed through. The current modal attachment list is horizontal, so don't need this one.
### What changes do you think we should make in order to solve the problem?

Removing prop `maxToRenderPerBatch` is a solution [here](https://github.com/Expensify/App/blob/22467c148c59d7a37c9e91dda089d70116270005/src/components/AttachmentCarousel/index.js#L364) .

### What alternative solutions did you explore? (Optional)
N/A

Here is a video link after fixed:
https://www.loom.com/share/bc5bb3fd5ec54a7daa52b405971688f1

**Comment by mkarkachov** [comment]
@lanitochka17 maybe this issue is related to this one https://github.com/Expensify/App/issues/21177

@univalchemist as I understand Removing prop `maxToRenderPerBatch` is setting it to default value. And after this there is that default Safari bug anyway. You can find it in your Loom video too. When you click on any image, 1st one displayed is image with cat (last image in your attachments) and then loaded correct image.

**Comment by univalchemist** [comment]
@mkarkachov , removing prop `maxToRenderPerBatch` is not setting it to the default value. It does not have any default value and removing it means `undefined`. Yes, this is just issue on Safari and it is related with this prop. Also notice that this prop affects in only the virtual list.

**Comment by CortneyOfstad** [comment]
Not overdue 👍

**Comment by kidroca** [comment]
The `maxToRenderPerBatch` parameter is not added by accident; the chosen value plays a crucial role. Despite the fact that we only see one item at a time, two additional items are rendered on each side (outside of view). This effectively preloads the subsequent and previous attachments, preparing the content that the user is likely to interact with next.

The list's orientation, horizontal in this case, does not provide a valid reason for eliminating this property. The `maxToRenderPerBatch` parameter is a vital part of virtualization, and it operates irrespective of the type of the list.

Furthermore, this property has a default value of `10` (see [maxToRenderPerBatchOrDefault](https://github.com/facebook/react-native/blob/72abed2c96d65769567e2b7e492764c1a58e6e81/packages/virtualized-lists/Lists/VirtualizedList.js#L98C1-L100)). This implies that removing the property would not solve the issue but merely displace it. It might also degrade loading as we'll start loading 10 images shortly after the carousel is opened

Erasing the property or augmenting the value of `maxToRenderPerBatch` only masks the issue. This is because the problem would then only surface for conversations encompassing more than 10 attachments.

**Comment by CortneyOfstad** [comment]
Not overdue

**Comment by univalchemist** [comment]
> The `maxToRenderPerBatch` parameter is not added by accident; the chosen value plays a crucial role. Despite the fact that we only see one item at a time, two additional items are rendered on each side (outside of view). This effectively preloads the subsequent and previous attachments, preparing the content that the user is likely to interact with next.
> 
> The list's orientation, horizontal in this case, does not provide a valid reason for eliminating this property. The `maxToRenderPerBatch` parameter is a vital part of virtualization, and it operates irrespective of the type of the list.
> 
> Furthermore, this property has a default value of `10` (see [maxToRenderPerBatchOrDefault](https://github.com/facebook/react-native/blob/72abed2c96d65769567e2b7e492764c1a58e6e81/packages/virtualized-lists/Lists/VirtualizedList.js#L98C1-L100)). This implies that removing the property would not solve the issue but merely displace it. It might also degrade loading as we'll start loading 10 images shortly after the carousel is opened
> 
> Erasing the property or augmenting the value of `maxToRenderPerBatch` only masks the issue. This is because the problem would then only surface for conversations encompassing more than 10 attachments.

Ok, let's see who suggests another solution then!

**Comment by AshenI99** [comment]
Contributor details
Your Expensify account email: asheniranda99@gmail.com
Upwork Profile Link: http://upwork.com/freelancers/asheniranda99

**Comment by AshenI99** [comment]
Contributor details
Your Expensify account email: asheniranda99@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~016480a0e1383ea286

**Comment by kidroca** [comment]
In response to: https://github.com/Expensify/App/issues/22318#issuecomment-1634256318
@AshenI99, we utilize the `pagingEnabled` property to facilitate the horizontal pagination of attachments, displaying them one at a time.

The existing documentation for this property might be somewhat terse, but it serves a key purpose: it allows for a portion of the view to be scrolled through. In our scenario, we scroll through exactly one page (equivalent to one ScrollView size), and this functions correctly most of the time.

What leads you to the conclusion that removing `pagingEnabled` would resolve the issue? Wouldn't this modification also disrupt the current paging behavior?

**Comment by AshenI99** [comment]
Hi @kidroca,

You are right about that paging behavior when it comes to mobile devices. So, if we completely remove that, it'll affect the paging behavior of the mobile devices. However, when it comes to non-touch screens, paging behavior won't be an issue as a user is switching through attachments with arrows. So, it'll only switch one image at a time. 

Therefore, we need to disable the `pagingEnabled` only when it is a non-touchscreen such as desktops and laptops. So it'll be as follows,

`pagingEnabled={this.canUseTouchScreen}`


https://github.com/Expensify/App/assets/73756207/8fa9ecdc-eb18-4cdf-afc0-4e8dbd94d28d

**Comment by AshenI99** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/22318#issuecomment-1634256318)

**Comment by kidroca** [comment]
@AshenI99
> You are right about that paging behavior when it comes to mobile devices. So, if we completely remove that, it'll affect the paging behavior of the mobile devices. However, when it comes to non-touch screens, paging behavior won't be an issue as a user is switching through attachments with arrows. So, it'll only switch one image at a time.

I appreciate your observations. However, I need a bit more clarification on the specific role of `pagingEnabled` in causing the issue at hand. You've pinpointed it as the root of the problem, yet an in-depth explanation is missing. 

Could you shed light on the exact purpose of `pagingEnabled` and why you believe it is only relevant to touch screens? Also, it's worth noting that even on touch screen devices, we still offer the arrow buttons to facilitate navigation. Wouldn't those users be susceptible to encountering the same bug as well?

From my perspective, this is beginning to appear like a bug with how `pagingEnabled` functions in react-native-web, or perhaps it's related to our particular blend of property settings in conjunction with `pagingEnabled`.

**Comment by AshenI99** [comment]
@kidroca 

Yes. I think you are right. It could be either bug with how `pagingEnabled` functions in react-native-web, or it could related to our particular blend of property settings in conjunction with `pagingEnabled`.

Also this issue is happening in mobile browsers too. That case it is happening when we use swipe action to swipe through images instead of using arrows.

**Comment by kidroca** [comment]
Seems the following react-native-web bug might be related
- https://github.com/necolas/react-native-web/issues/2026

The video provided in the issue description depicts a behavior with content jumping to the start of the list as the user scrolls to the right. I think it's similar to what happens in our case. The content seem to jump to the start of the virtual list

**Comment by CortneyOfstad** [comment]
@rushatgabhane Bump ^^^

**Comment by CortneyOfstad** [comment]
@rushatgabhane any update on the above?

**Comment by rushatgabhane** [comment]
Agree! It looks like it's an underlying issue with react native web https://github.com/necolas/react-native-web/issues/2026

**Comment by CortneyOfstad** [comment]
@rushatgabhane Do we think any of the proposals above could help with this, or do we still need some new ones?

**Comment by univalchemist** [comment]
See here?
No anyone suggests a good solution than [this](https://github.com/Expensify/App/issues/22318#issuecomment-1633537915)
I am sure there is not any good way to fix this issue than my proposal.

**Comment by CortneyOfstad** [comment]
Heading OoO until 8/14, so reassigning BZ to keep an eye on this. 

@rushatgabhane bump on the proposals above ^^^^

**Comment by rushatgabhane** [comment]
waiting for proposals

**Comment by MitchExpensify** [comment]
Is this still the latest on this one @rushatgabhane ?

> waiting for proposals to fix

https://github.com/necolas/react-native-web/issues/2026

**Comment by MitchExpensify** [comment]
Friendly bump on [this question](https://github.com/Expensify/App/issues/22318#issuecomment-1667806931) @rushatgabhane

**Comment by rushatgabhane** [comment]
@MitchExpensify yes! this is the latest - https://github.com/Expensify/App/issues/22318#issuecomment-1660967959

**Comment by CortneyOfstad** [comment]
Hi @MitchExpensify -- back from OoO so I can take this back over. Thanks for holding down the fort while I was gone 👍

**Comment by MitchExpensify** [comment]
Sounds gooood! Thanks, @CortneyOfstad

**Comment by CortneyOfstad** [comment]
No updates yet 👍

**Comment by CortneyOfstad** [comment]
@rushatgabhane Just checking to see where we're at with this — TIA!

**Comment by CortneyOfstad** [comment]
Still no proposals for the upstream issue 👍

**Comment by wojtus7** [comment]
Hey, I am from the Software Mansion expert's agency.

## Proposal

### Please re-state the problem that we are trying to solve in this issue.

On desktop browsers images skip out of control when going next/previous attachment when there is more that 5 attachments available. 

### What is the root cause of that problem?

Bug in react-native-web Flatlist implementation.

### What changes do you think we should make in order to solve the problem?

Since issue exist in reaxt-native-web library since 2021 and it doesn't look like anybody will take care of it anytime soon I think it's reasonable to create separate solution to solve the issue without FlatList, because not being able to go through attachments on web is pretty serious issue IMO, and when FlatList issue will be fixed we can always go back.

Since it only happens on Desktop Web we can make it conditional only for those platforms, while still using flatlist anywhere else.

Simple Carousel I wrote still preloads 5 images at the time and behaves pretty much the same as FlatList

https://github.com/Expensify/App/assets/42337257/561eb675-86e8-46e5-a26a-d3754777c8e2

### What alternative solutions did you explore? (Optional)

Tried to fix the bug in react-native-web but failed

**Comment by CortneyOfstad** [comment]
@rushatgabhane thoughts?

**Comment by CortneyOfstad** [comment]
Bump @rushatgabhane

**Comment by mountiny** [comment]
Yeah I think that we should try to make a fix upstream if its a valid bug which is repro in their demo app or a simple set up

at the same time I agree that this is a higher value bug so we should prefer fixing this earlier maybe even using some simple carousel

**Comment by wojtus7** [comment]
@rushatgabhane 

> @wojtus7 what's stopping us from using the solution on all platforms?

The problem here is that on desktop web it's pretty straight forward with only going previous and next, and some simple zoom capability onClick. With all platforms we have to take care  of some more complicated gesture handling, zooming, panning, swiping and all kind of interactions that intersect with each other.

Adding solution for all platforms would give a lot of complexity to the app we probably don't want to maintain, that's why my solution was barely ~30 lines of code only focusing on the broken part and being easy to understand and maintain without possibility for much edge cases and strange bugs.

But, yes - overall I agree that fixing the bug in react-native-web would be the perfect solution

**Comment by CortneyOfstad** [comment]
@rushatgabhane bump ^^^

**Comment by wojtus7** [comment]
Unfortunately as I was testing all different platforms before creating a PR I realised same thing happens not only for desktop web as initially suggested in the bug report but the mobile web is also broken so I removed `if` for desktop and used only simple carousel for all attachments on web.

Unfortunately as it removes swipes ability from mobile web, I don't know if my solution still makes as much sense since it's less comfortable to use being able to go next/previous only using buttons but for sure it makes this bug even more troublesome.

https://github.com/Expensify/App/assets/42337257/8ba0c318-679a-49e5-89f8-4e57dba4ed87

Created a PR this way or another so you can take a look at the code, but feel free to close it

**Comment by wojtus7** [comment]
If you want me to I can add swiping functionality to my carousel, if solves the problem here, but it will greatly increase complexity of the solution 

In the meantime I will go once again in react-native-web and try to find a solution inside the library with a fresh look but I'm afraid there's little chance I will succeed 🫠

This way or another I am eager to help solve this issue 🙏

**Comment by CortneyOfstad** [comment]
@joelbettner bump on the comments above

**Comment by CortneyOfstad** [comment]
Heading OoO until Oct. 25, so reassigning BZ to keep an eye on this while I am gone 👍

**Comment by CortneyOfstad** [comment]
@dylanexpensify — I'm back from OoO, so I can take this over again. Thanks for keeping an eye on things!

**Comment by dylanexpensify** [comment]
Wooot! Welcome back!

**Comment by mczernek** [comment]
> @dylanexpensify — I'm back from OoO, so I can take this over again. Thanks for keeping an eye on things!

Welcome back! I'd like to push that from SWM side :)

I believe we were blocked by waiting on WishList implementation in hopes that it would make matters better. Since this didn't happen I believe we should consider @wojtus7's PR as a valid solution for the problem.

I believe it is very serious issue since it makes attachment modal non usable on web for any conversation with more than 4 attachments (so virtually any active conversation)

**Comment by CortneyOfstad** [comment]
@joelbettner any thoughts on @mczernek's comment above? Thanks!

**Comment by kosmydel** [comment]
Hey, I'm Jakub from Software Mansion. 

Can I ask for any response to the @mczernek comment? Please let me know if we should proceed with the already prepared solution by @wojtus7. 

cc @CortneyOfstad

**Comment by kosmydel** [comment]
Hey, today I tried to reproduce it (both on main and on staging). But it looks like this issue doesn't occur anymore. Can someone test it and confirm, or am I doing anything wrong?

https://github.com/Expensify/App/assets/104823336/5952c8f5-7a3d-426b-97d2-538f4927d7b2

cc @CortneyOfstad, @rushatgabhane

**Comment by kidroca** [comment]
Replacing `FlatList` with `Flashlist` is also an option 

I think I've found a related ticket, fixing the current problem, by either coding our own minimal replacement, or using something other than `FlatList` would fix the other bug as well
- #31166

**Comment by joelbettner** [comment]
@kidroca if using `Flashlist` instead of `Flatlist` fixes the problem, I think that is the way to go.

**Comment by kidroca** [comment]
Hi team,

I've been following the discussion regarding the issue with switching attachments in Safari and the potential switch from `FlatList` to `Flashlist`. 

It seems we've come to an agreement that if possible we should use `Flashlist`. 

I'd like to propose conducting a test to evaluate `Flashlist` as a replacement. My aim is to rigorously test its functionality to ensure it resolves the current issue without introducing new ones. However, before proceeding, I need confirmation and approval from the team.

If given the green light, I plan to track this as an hourly item and provide the results here. This test will help us determine if `Flashlist` is the right way forward and potentially make the rest of the ticket actionable.

Looking forward to your feedback and hoping for the go-ahead to start this task.

**Comment by kosmydel** [comment]
Hey @kidroca, thanks for your answer!

I've consulted it with the SWM team, as we were previously responsible for this issue, and we have nothing against it (especially since we are busy with other issues).

cc @CortneyOfstad, @joelbettner, @rushatgabhane for final decision

**Comment by rushatgabhane** [comment]
sounds good! @joelbettner let's assign @kidroca to this issue then

**Comment by kidroca** [comment]
Hi, I hope you all had a wonderful holiday season!

A quick update: there haven't been any new developments due to the Christmas holidays.
I'll be resuming work and will keep you all informed of any updates or changes as soon as they occur.

Thank you for your understanding and patience.

**Comment by kidroca** [comment]
Hi team,

I've completed the evaluation of FlashList as a potential replacement for FlatList in our chat attachment viewer. The results are mixed: FlashList solves certain existing issues but introduces new challenges. For specific code changes, refer to the PR: [Expensify/App#34639](https://github.com/Expensify/App/pull/34639).

**Key Findings:**

1. **Resolution of Previous Issues**: FlashList effectively addresses several known problems:
   - **Image Skipping Issue**: Successfully resolves the problem of skipping images when switching attachments.
   - **Safari Image Opening**: Fixes the issue reported in [Issue #23546](https://github.com/Expensify/App/issues/23546).
   - **Protected PDF Viewing on Mobile Web**: Addresses the challenge detailed in [Issue #31166](https://github.com/Expensify/App/issues/31166).

2. **Introduction of New Challenges**:
   - **Snap-to-Interval Alignment**: On mobile, FlashList doesn't adequately support this feature, crucial for intuitively centering images in our carousel. The current implementation results in an abrupt stop of the scroll position, impacting the user experience.
   - **Reloading on View Resizing**: Mirroring an issue with FlatList (referenced in PR [#26543](https://github.com/Expensify/App/pull/26543)), FlashList needs to reload for size updates when the view is resized. The previously implemented fix for FlatList doesn't apply here.

3. **Beta Status of FlashList for Web**: The FlashList implementation for web is in beta, as noted in the [FlashList documentation](https://shopify.github.io/flash-list/docs/known-issues#4-web-support-is-in-beta). This may be contributing to the observed challenges.

I'd like to note recent changes to the attachment carousel code, presently we have a separate implementation for iOS and Android (`index.native.js`), with that in mind a simpler ScrollView implementation might be more apt for web and desktop, now that `index.js` serves only them.

**Proposal**: I suggest exploring the ScrollView option for web and desktop, anticipating a simplified and more efficient solution. 

Additionally, during this investigation, I've linked this issue to 3 other tickets with similar underlying problems. My findings are detailed in this comment: [Issue #23546 Comment](https://github.com/Expensify/App/issues/23546#issuecomment-1895677762).

I look forward to your thoughts on proceeding with the ScrollView approach.

**Comment by CortneyOfstad** [comment]
Thanks @kidroca! @joelbettner any thoughts ^^^ ?

**Comment by joelbettner** [comment]
@kidroca I think proceeding with the ScrollView approach sounds good.

**Comment by CortneyOfstad** [comment]
@kidroca any updates? Thanks!

**Comment by kidroca** [comment]
Currently focused on other urgent tasks, so I haven't made any progress yet. I plan to come back to this ticket by the end of the week. Thanks for your patience.

**Comment by sourcecodedeveloper** [comment]
@CortneyOfstad @joelbettner @rushatgabhane I do have proposal that might resolve the issue as I understand the issue is related to the AttachmentCarousel not loading the proper page index. I have proposed [solution](https://github.com/Expensify/App/issues/23546#issuecomment-1928979778) for issue #23546 which would potentially also apply to the current issue. The AttachmentCarousel page index bug is only appearing in safari and has also affected password protected pdfs(after entering the password, pdf with wrong page index is opened in the carousel) The #23546 has been opened since Jul 25, 2023 and I am curious to find out if what I have proposed resolves the safari related bug(s).

**Comment by CortneyOfstad** [comment]
@kidroca any updates?

Also, @sourcecodedeveloper, we have someone assigned already, but thank you for reaching out!

**Comment by kidroca** [comment]
Hi team,

I wanted to provide a quick status update on the ticket regarding the carousel bug we've been tackling. Despite implementing the ScrollView strategy I previously proposed, the issue persisted, leading me to conduct a deeper investigation. It became clear that the problem wasn't specific to the use of ScrollView or FlatList; rather, it was rooted in how Safari on iOS manages snap alignment when cycling through attachments. I've shared a [detailed analysis on a related ticket](https://github.com/Expensify/App/issues/23546#issuecomment-1957900164), given its relevance to the discussion there.

**Ticket History Summary:**
- My initial investigation pointed towards react-native-web as the culprit: https://github.com/Expensify/App/issues/22318#issuecomment-1637856598
- Current ticket was linked to several similar issues
    - https://github.com/Expensify/App/issues/23546#issuecomment-1895677762
- We experimented with replacing FlatList with FlashList, which introduced a different set of challenges: [FlashList Attempt Summary](https://github.com/Expensify/App/issues/22318#issuecomment-1895711136).
- Presently: Substituting FlatList with a plain ScrollView also failed to resolve the bug 

The core of the issue has been identified as CSS snap alignment of scroll items, as elaborated in my recent analysis
- https://github.com/Expensify/App/issues/23546#issuecomment-1957900164

I'm currently exploring alternatives and keeping an eye on proposals in related tickets. I'll ensure to keep everyone updated with any new developments or insights.

Thank you for your patience and collaboration as we work through this complex issue.

**Comment by CortneyOfstad** [comment]
Thank you @kidroca for the detailed explanation! This is incredibly thorough, and if there is anything that we can do to help in support, do not hesitate to ask! Within BZ, I will also keep an eye out for anything related that could help 👍

**Comment by kidroca** [comment]
Hi team,

I'm reaching out to share an important update on the carousel bug across several related tickets. I realized I might not have communicated this breakthrough in all relevant discussions, so I wanted to ensure everyone is on the same page.

Exciting news: I've found and proposed a straightforward solution that effectively tackles the attachment carousel issue and simultaneously addresses the related challenges we encountered. For a detailed account of the steps taken and the rationale behind them, please refer to my comprehensive summary here: [Issue Summary and Solution](https://github.com/Expensify/App/issues/23546#issuecomment-1961954931), which is currently pending a C+ review.

The essence of the solution lies in our ability to manually manage scroll and panning interactions, thereby circumventing the pitfalls associated with scroll snap alignment in virtual lists. This approach has shown promising results in our initial tests, offering a solid fix for the issues we've been diligently working to resolve.

Thank you for your continued patience and collaborative spirit as we navigate these challenges together. I'll keep you all updated with any further developments.

**Comment by CortneyOfstad** [comment]
Bumped the on-hold GH [here](https://github.com/Expensify/App/issues/23546#issuecomment-1999741830). Will follow-up as soon as they respond 👍

**Comment by CortneyOfstad** [comment]
Draft PR was created [here](https://github.com/Expensify/App/pull/39930)! 🎉

**Comment by CortneyOfstad** [comment]
PR is still in draft but there is a stale PR in the on-hold GH, so reaching out to Scott (engineer on the main GH) to help move this along [here](https://github.com/Expensify/App/issues/23546#issuecomment-2061263197)

**Comment by CortneyOfstad** [comment]
Just bumped the on-hold issue, as the PR is ready for review!

**Comment by CortneyOfstad** [comment]
Bumped [here](https://github.com/Expensify/App/issues/23546#issuecomment-2082759402)

**Comment by CortneyOfstad** [comment]
On the on-hold GH, it was indicated that the PR would be merged early next week — will continue to keep an eye on that 👍

**Comment by CortneyOfstad** [comment]
Bumped the on-hold GH to see if we're still on pace for this week — will follow up as soon as I hear back 👍

**Comment by CortneyOfstad** [comment]
Merge freeze in place, so having to wait a bit for the on-hold issue

**Comment by CortneyOfstad** [comment]
We're in a merge freeze right now, so this will be delayed a bit 👍

**Comment by CortneyOfstad** [comment]
Merge freeze ends today! 🤞

**Comment by CortneyOfstad** [comment]
Was deployed to production 4 days ago 😍

**Comment by CortneyOfstad** [comment]
Set to be out of the 7-day window tomorrow 🥳

**Comment by CortneyOfstad** [comment]
This can be closed as the on-hold issue has been resolved/completed! Wahooo!
